Example/Tests/Tests.m

```diff
@@ -934,8 +981,129 @@ - (ADBMediaObject *)createWithProperties:(NSDictionary *)properties andEventType
+
+                SEGMockPlaybackDelegateFactory *mockPlaybackDelegateFactory = [[SEGMockPlaybackDelegateFactory alloc] init];
+                mockPlaybackDelegateFactory.playbackDelegate = mockPlaybackDelegate;
+
+                SEGTrackPayload *payload = [[SEGTrackPayload alloc] initWithEvent:@"Video Quality Updated" properties:@{} context:@{}
+                    integrations:@{}];
+
+                [integration track:payload];
+                [verify(mockPlaybackDelegate) createAndUpdateQOSObject:@{}];
+            });
+        });
     });
+});
+describe(@"SEGPlaybackDelegate", ^{
+    __block SEGPlaybackDelegate *playbackDelegate;
+    beforeEach(^{
+        playbackDelegate = [[SEGPlaybackDelegate alloc] init];
+    });
+
+    describe(@"SEGPlaybackDelegate", ^{
+        it(@"sets initialTime on initialization", ^{
+            long currentTime = CFAbsoluteTimeGetCurrent();
+            assertThatLong(playbackDelegate.initialTime, equalToLong(currentTime));
+        });
 
+        it(@"gets incrementPlayheadPosition if not paused", ^{
+            playbackDelegate.isPaused = false;
+            playbackDelegate.offset = 0;
+            playbackDelegate.updatedPlayheadPosition = 0;
+            playbackDelegate.initialTime = 0;
+            long initialTime = CFAbsoluteTimeGetCurrent();
+            sleep(2);
+            assertThatLong(([playbackDelegate getCurrentPlaybackTime]), equalToLong((initialTime + 2) / 1000));
+        });
+
+        it(@"pauses playhead position if paused", ^{
+            playbackDelegate.isPaused = true;
+            [playbackDelegate getCurrentPlaybackTime];
+            assertThatLong(playbackDelegate.pausedPlayheadPosition, equalToLong(0));
+        });
+
+        it(@"increments playheadPosition", ^{
+            long currentTime = CFAbsoluteTimeGetCurrent();
+            playbackDelegate.offset = 0;
+            playbackDelegate.updatedPlayheadPosition = 0;
+            playbackDelegate.initialTime = 0;
+            [playbackDelegate incrementPlayheadPosition];
+            sleep(2);
+            assertThatLong(playbackDelegate.playheadPosition, equalToLong((currentTime + 2) / 1000));
+        });
+
+        it(@"pausePlayhead", ^{
+            [playbackDelegate pausePlayhead];
+            long currentTime = CFAbsoluteTimeGetCurrent();
+            assertThatLong(playbackDelegate.pausedPlayheadPosition, equalToLong(playbackDelegate.playheadPosition));
+            assertThatLong(playbackDelegate.pausedStartedTime, equalToLong(currentTime));
+            assertThatBool(playbackDelegate.isPaused, isTrue());
+        });
+
+        it(@"unPausePlayhead with no offset", ^{
+            playbackDelegate.offset = 0;
+            [playbackDelegate unPausePlayhead];
+            long currentTime = CFAbsoluteTimeGetCurrent();
+            playbackDelegate.pausedStartedTime = 0;
+            assertThatLong(playbackDelegate.offset, equalToLong(currentTime / 1000));
```

Comment:
What is this test checking? Shouldn't it be checking against the time before calling `unPausePlayhead`?